data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,558 @@
+---
+id: values-and-functions
+title: Values and Functions
+description: |
+  TBD
+category: "Language"
+---
+
+# Values and Functions
+
+## Introduction
+
+### Learning Goals
+
+When presenting OCaml or another functional programming language, it is often said: “Functions are treated as first-class citizens.” Without further explanation or context, this may not be helpful (it wasn't to me). The goal of this tutorial is to acquire the capabilities implied and entailed by this assertion. In turn, this should explain it:
+- Write all kinds of definitions, for all kinds of values, in all different ways
+- Accurately assess the scope of any definition
+- Avoid being fooled by shadowed definitions and scopes
+- Explain what a closure is and why it is so named
+- Partially apply functions, create closures, and pass them as values
+- Curry and uncurry functions
+- Use the `unit` type as a function parameter or return value, when needed
+
+### Chatting With OCaml
+
+This tutorial is shown using the UTop toplevel. Although other toplevels may be used, we recommend UTop. Code samples are intended to be copied/pasted into UTop. It would benefit the reader to write variations around the examples provided to strengthen understanding. The topics discussed are not limited to interactive execution of OCaml expressions. However, we believe they are easier to understand within the dynamics of interaction with the OCaml interpreter.
+
+### Touch Base on Double Semicolon
+
+When using UTop to interact with the OCaml interpreter, lines ending with double semicolons trigger the parsing, type-checking, and evaluation of everything typed between the prompt and the double semicolon. This may span several lines. The interpretation of that double semicolon isn't made by the OCaml interpreter; it is made by UTop, the OCaml toplevel. Once the evaluation of a double semicolon terminated entry is over, the REPL waits for another piece of input.
+
+Nevertheless, the double semicolon `;;` is a valid token in the OCaml syntax. It is a [no-op](https://en.wikipedia.org/wiki/NOP_(code)), i.e., it does not trigger any behaviour. It is ignored by the compiler. In OCaml, source code files meant to be compiled or interpreted as scripts, double semicolons can and should be avoided. This is a means to avoid double semicolons required by the toplevel to raise errors when using the compiler.
+
+## Non-Function Values
+
+### Values
+
+Like most functional programming languages, OCaml is an [expression-oriented programming language](https://en.wikipedia.org/wiki/Expression-oriented_programming_language). There are no statements, i.e., syntactical constructions made to produce some sort of computation. Jumps are examples of statements. Computations triggered by OCaml are written as expressions. Once completed, they produce a value, which has a type. Here are a few examples of expressions, their type, and resulting values.
```

Comment:
Are statements "syntactical constructions made to produce some sort of computation" or are statements "syntactical constructions" that are "made to produce some sort of computation"?

"Jumps are examples of statements" seems to lack some context. What are Jumps (other than examples of statements) and how does it relate to the following sentence?

Reply:
> Like most functional programming languages, OCaml is an [expression-oriented programming language](https://en.wikipedia.org/wiki/Expression-oriented_programming_language). There are no [statements](https://en.wikipedia.org/wiki/Statement_(computer_science)), i.e., syntactical constructions made to produce some change of state. Variable [assignment](https://en.wikipedia.org/wiki/Assignment_(computer_science)) or [for loop](https://en.wikipedia.org/wiki/For_loop) are examples of statements. Computations triggered by OCaml are written as expressions. Once completed, they produce a value, which has a type. Here are a few examples of expressions, their type, and resulting values.